EIPS/eip-is-contract.md

```diff
@@ -0,0 +1,106 @@
+---
+title: IS_CONTRACT instruction
+description: Introduce IS_CONTRACT instruction
+author: Andrei Maiboroda (@gumb0), Piotr Dobaczewski (@pdobacz), Danno Ferrin (@shemnon)
+discussions-to: <URL>
+status: Draft
+type: Standards Track
+category: Core
+created: 2024-09-01
+requires: 3540, 7692
+---
+
+## Abstract
+
+Allow EOF contracts to discriminate between EOA (Externally Owned Accounts) and smart contract accounts by introducing an `IS_CONTRACT` instruction.
```

Comment:
```suggestion
Allow EOF contracts to discriminate between EOA (Externally Owned Accounts) and contracts (including smart contract accounts) by introducing an `IS_CONTRACT` instruction.
```

Reply:
@abcoathup Actually, we'd like a clarification of your suggestion. "smart contract accounts" was a mental shortcut intending to mean "EVM state accounts with code", and probably that wording is clearer and more specific. Next, maybe we change "smart contract accounts" to "smart contract wallets" to avoid confusion. Would this work?

Reply:
This is what I have in mind: 2be26b34 - I think "contract account" is a well established and simple term, hope this works.